e2e-polybft/e2e/jsonrpc_test.go

```diff
@@ -302,11 +301,73 @@ func TestE2E_JsonRPC(t *testing.T) {
 		require.True(t, txn.Succeed())
 		txReceipt := txn.Receipt()
 
-		var header types.Header
-		err = newEthClient.EndpointCall("eth_getHeaderByHash", &header, txReceipt.BlockHash)
+		header, err := newEthClient.GetHeaderByHash(types.Hash(txReceipt.BlockHash))
 		require.NoError(t, err)
 
 		require.Equal(t, txReceipt.BlockNumber, header.Number)
-		require.Equal(t, txReceipt.BlockHash, ethgo.Hash(header.Hash))
+		require.Equal(t, types.Hash(txReceipt.BlockHash), header.Hash)
+	})
+
+	t.Run("eth_getBlockReceipts", func(t *testing.T) {
+		txn := cluster.Transfer(t, preminedAcct, types.StringToAddress("0xDEADBEEF"), one)
+		require.True(t, txn.Succeed())
+		receipt := txn.Receipt()
+
+		receipts, err := newEthClient.GetBlockReceipts(bladeRPC.BlockNumber(receipt.BlockNumber))
+
+		require.NoError(t, err)
+		require.NotNil(t, receipts)
+		require.NotEqual(t, 0, len(receipts))
+
+		require.Equal(t, receipts[0].BlockNumber, receipt.BlockNumber)
+		require.Equal(t, receipts[0].BlockHash, receipt.BlockHash)
+	})
+
+	t.Run("eth_createAccessList", func(t *testing.T) {
+		deployTxn := cluster.Deploy(t, preminedAcct, contractsapi.TestSimple.Bytecode)
+		require.True(t, deployTxn.Succeed())
+		receipt := deployTxn.Receipt()
+
+		target := types.Address(receipt.ContractAddress)
+		input := contractsapi.TestSimple.Abi.GetMethod("getValue").ID()
+
+		gasPrice, err := newEthClient.GasPrice()
+		require.NoError(t, err)
+
+		// setValueFn := contractsapi.TestSimple.Abi.GetMethod("setValue")
```

Comment:
Remove this comment if not needed.

Reply:
The idea was to send a transaction that would set the value to some non-zero value, and at the end to assert that access list result contain storage key for the `value` field as well.

Something like this, but for Input, provided the result of `setValueFn.Encode(...)` function

```go
txn := types.NewTx(
			types.NewLegacyTx(
				types.WithFrom(newAccountAddr), // adapt
				types.WithTo(&target),
				types.Input(...), //setValueEncoded byte array,
			))
	
		signedTxn, err := crypto.NewLondonSigner(chainID.Uint64()).SignTx(txn, newAccountKey) // use the proper account key
		require.NoError(t, err)
		data := signedTxn.MarshalRLPTo(nil)
		hash, err := newEthClient.SendRawTransaction(data)
		require.NoError(t, err)
		require.NotEqual(t, types.ZeroHash, hash)
```